src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidTestCommand.cs

```diff
@@ -120,6 +143,36 @@ protected override Task<ExitCode> InvokeInternal(ILogger logger)
             return Task.FromResult(ExitCode.GENERAL_FAILURE);
         }
 
+        private string GetDeviceToUse(ILogger logger, AdbRunner runner, string apkRequiredArchitecture)
+        {
+            var allDevicesAndTheirArchitectures = runner.GetAttachedDevicesAndArchitectures();
+            if (allDevicesAndTheirArchitectures.Count == 1)
+            {
+                // There's only one device, so -s argument isn't needed but still check
+                if (!allDevicesAndTheirArchitectures.Values.Contains(apkRequiredArchitecture, StringComparer.InvariantCultureIgnoreCase))
+                {
+                    logger.LogError($"Single device available has architecture '{allDevicesAndTheirArchitectures.Values.First()}', != '{apkRequiredArchitecture}'. Package installation will likely fail, but we'll try anyways."); 
+                }
+                return ""; // Empty = Use default device
```

Comment:
```suggestion
                return string.Empty; // Empty = Use default device
```

Reply:
But again, I think it would be better to utilize the compiler, allow `string?` on return and return `null` and handle that later. The API of the method then also becomes more intuitive since you can probably guess it returns `null` when no device found and does not for example throw an exception.